### PR TITLE
Support JDK10+ version-string format

### PR DESCRIPTION
**Pull request by benoitdesire**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -133,19 +133,21 @@ subprojects {
...
+                logging.level = LogLevel.INFO
+                taskdef(name: 'javancss', classname: 'javancss.JavancssAntTask', classpath: configurations.javancss.asPath)
+                javancss(
+                        srcdir: 'src/main/java',
+                        abortOnFail: 'true',
+                        generateReport: 'true',
+                        packageMetrics: 'true',
+                        classMetrics: 'false',
+                        functionMetrics: 'false'
+                        // outputFile: reportName,
```
Comment (philsss) [inline build.gradle:148]: to remove ?

File: build.gradle [hunk 2]
```diff
@@ -249,7 +251,7 @@ task jacocoRootReport(type: org.gradle.testing.jacoco.tasks.JacocoReport) {
         html.enabled = true
         xml.enabled = true
         csv.enabled = false
-        xml.destination = "${buildDir}/reports/jacoco/test/jacocoTestReport.xml"
+        //xml.destination = "${buildDir}/reports/jacoco/test/jacocoTestReport.xml"
```
Comment (philsss) [inline build.gradle]: same here ?

**Review by philsss — APPROVED** [review]

**Review by philsss — APPROVED** [review]

File: capsule/src/main/java/Capsule.java [hunk 3]
```diff
@@ -4368,19 +4368,27 @@ private static boolean equals(int[] a, int[] b, int n) {
         return true;
     }
 
-    private static final Pattern PAT_JAVA_VERSION = Pattern.compile("(?<major>\\d+)(\\.(?<minor>\\d+))?(?:\\.(?<patch>\\d+))?(_(?<update>\\d+))?(-(?<pre>[^-]+))?(-(?<build>.+))?");
+    private static final Pattern PAT_JAVA_VERSION = Pattern.compile("(?<major>\\d)(\\.(?<minor>\\d+))?(?:\\.(?<patch>\\d+))?(_(?<update>\\d+))?(-(?<pre>[^-]+))?(-(?<build>.+))?");
+    // from https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Runtime.Version.html
+    private static final Pattern PAT_JAVA_VERSION_JDK10 = Pattern.compile("(?<major>\\d+)(\\.(?<minor>\\d+))?(?:\\.(?<patch>\\d+))?(\\.(?<update>\\d+))?(-(?<pre>[^-]+))?.*");
```
Comment (TheItivitist) [inline capsule/src/main/java/Capsule.java:4376]: isn't it the contrary, update first then patch?


if I sum up, we have those string version formats:
- 1.major.minor_update from java 1.3 to 1.8
- major.minor.update from java 9
- major.minor.update.patch from java 10+

I think the java 9 format is way closer to java 10 format than java 1.3-8, except we will never have the patch part.
So we could have the first regex parsing "1.major.minor_update" up to java 8, and the second "major.minor.update.patch" from java 9 onward?

**Review by TheItivitist — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (benoitdesire) [reply build.gradle:148]: Since i haven't commented it, i prefer to keep it like that

File: build.gradle (hunk 2 above)
Comment (benoitdesire) [reply build.gradle]: Uncommented and fixed the compilation against last Gradle, thanks for notifying

File: capsule/src/main/java/Capsule.java (hunk 3 above)
Comment (benoitdesire) [reply capsule/src/main/java/Capsule.java:4376]: Yes ok, done

**Review by TheItivitist — APPROVED** [review]

**Pull request merged**